File: backend/src/main/kotlin/com/backend/backend/services/QuoteItemService.kt
package com.backend.backend.services

import com.backend.backend.domain.QuoteItem
import com.backend.backend.model.ProductDTO
import com.backend.backend.model.QuoteItemCharacteristicDTO
import com.backend.backend.model.QuoteItemDTO
import reactor.core.publisher.Flux
import reactor.core.publisher.Mono

interface QuoteItemService {
    fun addOne(quoteId: Int, product: ProductDTO): Mono<QuoteItem>
    fun addAll(quoteId: Int, quoteItems: List<QuoteItemDTO>): Mono<List<QuoteItem>>
    fun updateCharacteristics(
        quoteItemId: Int,
        characteristics: List<QuoteItemCharacteristicDTO>
    ): Flux<QuoteItemCharacteristicDTO>

    fun delete(quoteItemId: Int): Mono<Void>
}

File: frontend/src/services/quote/interface/QuoteItemService.ts
import {Characteristic} from "../../../model/product.model";
import {Quote} from "../../../model/quote.model";

export abstract class QuoteItemService {
    protected readonly url: string;

    constructor(url: string) {
        this.url = url;
    }

    abstract updateCharacteristics(quoteItemId: number, characteristics: Characteristic[]): Promise<Quote | undefined>
}